src/main/java/com/groupseven/sysc4806project/BookstoreController.java

```diff
@@ -39,4 +42,53 @@ public String home(
         model.addAttribute("books", books);
         return "index";
     }
+
+    @PostMapping("/add-book")
+    public String add(
+            @RequestParam(defaultValue = "") String isbn,
+            @RequestParam(defaultValue = "") String title,
+            @RequestParam(defaultValue = "") String description,
+            @RequestParam(defaultValue = "") String author,
+            @RequestParam(defaultValue = "") String publisher,
+            @RequestParam(defaultValue = "") Integer inventory,
+            @RequestParam(required = false) MultipartFile image
+    ) {
+        if (inventory < 0) {
+            inventory = 0;
+        }
+        this.bookService.create(isbn,title,description,author,publisher,inventory,image);
+        return "redirect:/home";
+    }
+
+    @PostMapping("/edit-book")
+    public String edit(
+            @RequestParam(required = false) String isbn,
+            @RequestParam(required = false) String title,
+            @RequestParam(required = false) String description,
+            @RequestParam(required = false) String author,
+            @RequestParam(required = false) String publisher,
+            @RequestParam(required = false) Integer inventory,
+            @RequestParam(required = false) MultipartFile image,
+            @RequestParam Integer book_id
+    ) {
+        if (inventory < 0) {
+            inventory = 0;
+        }
+        this.bookService.update(book_id,isbn,title,description,author,publisher,inventory,image);
+        return "redirect:/home";
```

Comment:
just a question, is redirect necessary? would it still work with using `return "index"`

Reply:
`return "index"` would require populating the Model for the page again. Using the redirect allows us to reuse the function to render the homepage

Reply:
not quite since if you just return index, it would just return the page without any book on it since /home is the thing that renders the book to index.

Reply:
ah kk I see, will def use this for the cart page ty!